File: generated_code/python/solution_3684_2.py
class Solution:
    def hasMatch(self, s: str, p: str) -> bool:
        # Since p contains exactly one '*', we can split it into a prefix and a suffix.
        # The '*' can match any sequence of zero or more characters.
        # Thus, p can match a substring of s if and only if we can find the prefix in s,
        # and then find the suffix in the remaining part of s after the prefix's match.
        prefix, suffix = p.split('*')
        
        # Find the first occurrence of the prefix in s
        prefix_idx = s.find(prefix)
        if prefix_idx == -1:
            return False
            
        # Find the suffix in the portion of s that comes after the end of the matched prefix
        # s.find(suffix, start_pos) searches for suffix starting from index start_pos
        suffix_idx = s.find(suffix, prefix_idx + len(prefix))
        
        return suffix_idx != -1

File: generated_code/cpp/solution_3684_2.cpp
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool hasMatch(string s, string p) {
        // Find the single '*' character in the pattern string p
        size_t star_idx = p.find('*');
        
        // Split the pattern into a prefix and a suffix around the '*'
        string prefix = p.substr(0, star_idx);
        string suffix = p.substr(star_idx + 1);
        
        // Find the first occurrence of the prefix in string s
        size_t prefix_idx = s.find(prefix);
        if (prefix_idx == string::npos) {
            return false;
        }
        
        // The suffix must appear after the end of the matched prefix.
        // Since '*' can match 0 or more characters, the suffix can start 
        // immediately where the prefix ends (prefix_idx + prefix.length()).
        size_t suffix_idx = s.find(suffix, prefix_idx + prefix.length());
        
        // If both prefix and suffix are found in the correct order, it's a match
        return suffix_idx != string::npos;
    }
};